### PR TITLE
Move Mac testing to commit only

### PR DESCRIPTION
**Pull request by jaredpar**
After some investigation into machine count and overlap we've decided to move the Mac testing to occur on commits only.  This means it will no longer appear on the PR queue.  The justification is as follows:

- Mac machines are a limited resource because it must be physical hardware.  Can't just spin up more Mac VMS in Azure as we please.  Hence we must be a bit more judicious with our usage.
- 83% of all Mac jobs in Roslyn were PR jobs.  About 1/3 of these were actually dead builds.  Essentially builds queued / running where PR had a more recent push.  Hence these Mac builds never contributed to PR validation and were just eating resources.
- There is incredible overlap between our Linux and Mac jobs.  In the 9 months we've been running both jobs there have been 2 failures which only one of the jobs would have caught.  At least one of them would have been marked for explicit testing on both OS.

In short we're paying a lot for this right now and getting very litte in return.

Mac will **still** be validated though.  Every commit to master or future will trigger a Mac job and the results will be actively monitored by the infrastructure team.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @davkean, @tannergooding, @dotnet/roslyn-infrastructure 

@tannergooding i don't know of any way to test this / deploy it to Jenkins.  Pretty sure the change is correct though.

File: netci.groovy [hunk 1]
```diff
@@ -155,7 +155,8 @@ static void addPullRequestTrigger(def myJob, String contextName, String opsysNam
   ['win', 'linux', 'mac'].each { opsys ->
     // folder("${branchName.substring(0, 6)}/${opsys.substring(0, 3)}")
     ['dbg', 'rel'].each { configuration ->
-      if ((configuration == 'dbg') || ((branchName != 'prtest') && (opsys == 'win'))) {
+      if (((configuration == 'dbg') && (branchName != 'prtest' || opsys != 'mac')) ||
```
Comment (agocke) [inline netci.groovy]: I think a comment explaining this conditional would be useful.

**Comment by davkean** [comment]
Before we do this can we wait for this weeks results? The MacPro's are now taking requests, and things are looking good.

**Comment by agocke** [comment]
:+1:

**Comment by TyOverby** [comment]
:+1:

File: netci.groovy (hunk 1 above)
Comment (davkean) [inline netci.groovy]: Agreed, I find it very hard to read.

**Comment by davkean** [comment]
Okay, assuming the comment that Andy and I mentioned, I give up, let's merge this in. 

Between us being down Mac machines and the mono crashes on Mac, this queue is giving us significantly more noise than benefit.

**Comment by davkean** [comment]
@dotnet-bot retest prtest/win/tst/unit64 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_tst_unit64/106/
// Retest reason:

**Comment by jaredpar** [comment]
@davkean let me rework this as I changed this script quite a bit since originally sending this PR.

**Comment by jaredpar** [comment]
@davkean, @agocke, @TyOverby 

Please rereview as I significantly changed this.

**Comment by agocke** [comment]
:+1:

**Comment by jaredpar** [comment]
@davkean let me know when you want me to pull the trigger on this.  I'm going to be leaving a bit early today.  May be better to wait till tomorrow AM so I can help clean up anything that goes wrong.

**Comment by TyOverby** [comment]
LG™
